### DISCUSSION TITLE
[Expo Router] How to do SSR with API routes?

### DISCUSSION THREAD

**Discussion by jbains90** [expo/expo#29033] [Archived: Questions]
Hi there!

Nice work on the API routes integration, I know it's still in beta but I'm keen to test it out on our new project. 

I looked at your docs [here](https://docs.expo.dev/router/reference/api-routes/) and I can't figure out if API routes are just designed to return JSON rather than HTML. It looks like the design structure is client focused with calls to a local backend for data but I'd like to make our app so that every API route serves HTML like a classic SSR setup.

Is this possible with API routes or if not, will it be planned for a future release?

**Comment by chucklapointe** [comment]
Any updates on this?

**Comment by rommyarb** [comment]
👉 [Using React Server Components in Expo Router apps](https://docs.expo.dev/guides/server-components)

**Reply by chucklapointe** [reply]
I have **RSC(Web) Bundling failed**  bundling issues as soon as I try to enable and run on Web.

```
experiments: {
      reactServerFunctions: true,
      reactServerComponentRoutes: true
    },
```

I'm running  Expo 53 and I have output: "server"

```
web: {
      output: "server",
      bundler: "metro",
    },
```

From what I'm understanding from the docs:

> Server Rendering RSC payloads to HTML is not supported yet. This means static and server output doesn't fully work yet.

Am I missing something?
